qiskit/optimization/algorithms/slsqp_optimizer.py

```diff
@@ -14,20 +14,69 @@
 
 """The SLSQP optimizer wrapped to be used within Qiskit's optimization module."""
 import logging
-from typing import List, cast
+from typing import List, cast, Tuple, Any, Union, Optional
 
 import numpy as np
 from scipy.optimize import fmin_slsqp
 
 from .multistart_optimizer import MultiStartOptimizer
 from .optimization_algorithm import OptimizationResult
 from ..exceptions import QiskitOptimizationError
+from ..problems import Variable
 from ..problems.constraint import Constraint
 from ..problems.quadratic_program import QuadraticProgram
 
 logger = logging.getLogger(__name__)
 
 
+class SlsqpOptimizationResult(OptimizationResult):
+    """
+    SLSQP optimization result, defines additional properties that may be returned by the optimizer.
+    """
+    def __init__(self, x: Union[List[float], np.ndarray], fval: float, variables: List[Variable],
+                 fx: Optional[np.ndarray] = None, its: Optional[int] = None,
+                 imode: Optional[int] = None, smode: Optional[str] = None) -> None:
+        """
+        Constructs a result object with properties specific to SLSQP.
+
+        Args:
+            x: The solution of the problem
+            fval: The value of the objective function of the solution
+            variables: A list of variables defined in the problem
+            fx: The value of the objective function being optimized, may be different from ``fval``
+            its: The number of iterations.
+            imode: The exit mode from the optimizer
+                (see the documentation of ``scipy.optimize.fmin_slsqp``).
+            smode: Message describing the exit mode from the optimizer.
+        """
+        super().__init__(x, fval, variables, None)
+        self._fx = fx
+        self._its = its
+        self._imode = imode
+        self._smode = smode
+
+    # pylint:disable=invalid-name
+    @property
+    def fx(self) -> np.ndarray:
```

Comment:
As`fx` is Optional in constructor the return should be Optional[int] here too since it can return None. Same comment to `its`, `imode` and `smode`

Reply:
Done